activesupport/lib/active_support/i18n_railtie.rb

```diff
@@ -4,6 +4,15 @@
 
 module I18n
   class Railtie < Rails::Railtie
+    module Helpers
+      def self.watched_dirs_with_extensions(paths)
+        paths.inject({}) do |dirs, path|
```

Comment:
Can we use `each_with_object`? I think that's closer to the intent of what we're doing (and I think it's a bit faster too).

```ruby
paths.each_with_object({}) do |path, result|
  result[path.absolute_current] = path.extensions
end
```